### ISSUE TITLE
[Webpack] Module not found: Error: Can't resolve 'vertx' in '/home/user/app/node_modules/plotly.js/dist'

### ISSUE THREAD

**Issue by brussee** [plotly/plotly.js#2466]
Error with plotly.js 1.35.1 and webpack 3.11.0 when importing: `import Plotly from 'plotly.js/dist/plotly-with-meta'`
```
./node_modules/plotly.js/dist/plotly-with-meta.js
Module not found: Error: Can't resolve 'vertx' in '/home/user/app/node_modules/plotly.js/dist'
 @ ./node_modules/plotly.js/dist/plotly-with-meta.js 61621:16-26
 @ ./src/entryPoint.js
 @ multi ./src/entryPoint.js webpack-hot-middleware/client?path=http://localhost:3000/__webpack_hmr
```

**Comment by brussee** [comment]
Exact same issue with `./node_modules/plotly.js/dist/plotly.min.js` and `./node_modules/plotly.js/dist/plotly.js` as well.
`Module not found: Error: Can't resolve 'vertx' in '/home/user/app/node_modules/plotly.js/dist'`

**Comment by nicolaskruchten** [comment]
Just to confirm: this is showing up as a warning, and compilation proceeds, right?

**Comment by chiaramdelucia** [comment]
@nicolaskruchten I am having the same issue and yes, it is logging as a `warning` and compilation proceeds.

**Comment by chiaramdelucia** [comment]
But, to note, I am now getting too many WebGL contexts in the console, where as before I wasn't. Not sure if this is related, but it did pop up with the 1.35.1 and this warning.

**Comment by etpinard** [comment]
> Not sure if this is related, but it did pop up with the 1.35.1 and this warning.

That shouldn't be related. @chiaramdelucia would you mind sharing a reproducible example? Thank you!

**Comment by chiaramdelucia** [comment]
@etpinard I didn't really think it would be, but thought I would note it JIC. 

As for reproducible code, here is a [codepen](https://codepen.io/chiaramdelucia/pen/ppMdOb?editors=1010). This isn't *exactly* what happens in my app, but it is similar. In the pen's console you'll see the following on the first button click, but not after that.
`Error: WebGL warning: Exceeded 16 live WebGL contexts for this principal, losing the least recently used one.` 

In my app, which uploads a dataset, it prints the following on the third time a dataset is uploaded. 
`WARNING: Too many active WebGL contexts. Oldest context will be lost.`

**Comment by etpinard** [comment]
Hmm. I can't replicate. Can anyone else try out @chiaramdelucia 's [codepen](https://codepen.io/chiaramdelucia/pen/ppMdOb?editors=1010)?

**Comment by chiaramdelucia** [comment]
Oh, wow. Now I can't reproduce it either with that codepen. I'm sorry about that. I guess it must be something specific to application setup. Thanks for looking @etpinard.

**Comment by tsaiDavid** [comment]
I'm actually experiencing the same issue - 

```
WARNING in ./node_modules/plotly.js/dist/plotly.js
Module not found: Error: Can't resolve 'vertx' in '/Users/***/node_modules/plotly.js/dist'
```

**Comment by hugograf** [comment]
Since v1.35.x, I'am also get a huge warning. It is used in Angular (angular-cli) project:
```
WARNING in ./node_modules/plotly.js/dist/plotly.js
Module not found: Error: Can't resolve 'vertx' in 'C:\SoftwareProjekte\Hugos\grafioschtrader\grafioschtraderClient\node_modules\plotly.js\dist'
resolve 'vertx' in 'C:\SoftwareProjekte\Hugos\grafioschtrader\grafioschtraderClient\node_modules\plotly.js\dist'
  Parsed request is a module
  using description file: C:\SoftwareProjekte\Hugos\grafioschtrader\grafioschtraderClient\node_modules\plotly.js\package.json (relative path: ./dist)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: C:\SoftwareProjekte\Hugos\grafioschtrader\grafioschtraderClient\node_modules\plotly.js\package.json (relative path: ./dist)
    resolve as module
...
````

**Comment by etpinard** [comment]
The webpack issue has been confirmed. No need for anyone test it themselves. https://github.com/plotly/plotly.js/issues/2466#issuecomment-371904885 were referenring to another issue.  Thanks.

**Comment by brussee** [comment]
> Just to confirm: this is showing up as a warning, and compilation proceeds, right?

Yes, webpack presents it as a warning and is able to complete the compilation.

**Comment by domehead100** [comment]
The error is from the es6 promise shim.  Discussed here [#100](https://github.com/stefanpenner/es6-promise/issues/100) and here  [#305](https://github.com/stefanpenner/es6-promise/issues/305).

I had to add 

new webpack.IgnorePlugin(/vertx/) 

to my plugins to squelch the warning.  Other things mentioned in the two discussions did not work.  I'm still on webpack 3.10 and started getting this when updating to plotly.js 1.35.2 a few days ago.

**Comment by elshuku** [comment]
experiencing the same issue during compile time.  Application and Plotly components all work fine.  For me it comes with another warning, which may or may not be related.

``

**Comment by domehead100** [comment]
It is a prebuilt file.  My recollection is that Uglify, which is smarter than webpack, was loading a module from es6 shim that was not transpiled, causing it to choke in release builds as soon as it encountered an es6 keyword such as let or const, or something like that.  To mitigate this, react-plotly.js was tweaked to load the prebuilt file, which is all in es5.  What bundler and version are you using?

**Comment by etpinard** [comment]
We're still hoping someone finds a fix from within plotly.js for this issue. Looks like that `new webpack.IgnorePlugin(/vertx/)` trick is working for some users as presented in https://github.com/plotly/plotly.js/issues/2466#issuecomment-372924684, but we should do better.

That said, that `es6-promise` polyfill will be on its way out in [`v2`](#420).

**Comment by etpinard** [comment]
also related: https://github.com/plotly/plotly.js/issues/2400

**Comment by nicolaskruchten** [comment]
I've updated https://github.com/plotly/plotly-webpack to document the fact that the `IgnorePlugin` workaround is the recommended fix for now.

**Comment by jackparmer** [comment]
This issue has been tagged with `NEEDS SPON$OR`

A [community PR](https://github.com/plotly/plotly.js/pulls?q=is%3Apr+is%3Aopen+label%3A%22type%3A+community%22) for this feature would certainly be welcome, but our experience is deeper features like this are difficult to complete without the Plotly maintainers leading the effort.

**Sponsorship range: $5k-$10k**

What Sponsorship includes:
- Completion of this feature to the Sponsor's satisfaction, in a manner coherent with the rest of the Plotly.js library and API
- Tests for this feature
- Long-term support (continued support of this feature in the latest version of Plotly.js)
- Documentation at [plotly.com/javascript](https://plotly.com/javascript/)
- Possibility of integrating this feature with Plotly Graphing Libraries (Python, R, F#, Julia, MATLAB, etc)
- Possibility of integrating this feature with Dash
- Feature announcement on [community.plotly.com](https://community.plotly.com/tag/announcements) with shout out to Sponsor (or can remain anonymous)
- Gratification of advancing the world's most downloaded, interactive scientific graphing libraries (>50M downloads across supported languages)

Please include the link to this issue when [contacting us](https://plotly.com/get-pricing/) to discuss.

**Comment by archmoj** [comment]
This might be fixed by #5358.

**Comment by alexcjohnson** [comment]
Yes, @archmoj feel free to close this issue.
